src/solver/type_getters.jl

```diff
@@ -865,28 +866,35 @@ function get_simulation(config::AtmosConfig)
     end
     @info "Prepared diagnostic callbacks: $s"
 
-    # It would be nice to just pass the callbacks to the integrator. However, this leads to
-    # a significant increase in compile time for reasons that are not known. For this
-    # reason, we only add one callback to the integrator, and this function takes care of
-    # executing the other callbacks. This single function is orchestrate_diagnostics
+    diagnostic_callbacks = tuple()
```

Comment:
@Sbozzolo, can we also skip the work of `diagnostics_iterations` above and below? Since those will not be relevant in the case of absent diagnostics?

Reply:
Also, get we remove the `try-catch` with `write_field!` in it and instead just call the callback. This will be simpler, and it will have a better chance of accurately precompiling the code we want.

Reply:
Same with the compute (just call the callback)

Reply:
Also, the block inside `if config.parsed_args["warn_allocations_diagnostics"]` doesn't really make sense, the contents will always allocate on the first call.

Reply:
> @Sbozzolo, can we also skip the work of `diagnostics_iterations` above and below? Since those will not be relevant in the case of absent diagnostics?

I changed to skip the work above. The work below is automatically skipped when there's 0 diagnostics.

> Also, get we remove the try-catch with write_field! in it and instead just call the callback. This will be simpler, and it will have a better chance of accurately precompiling the code we want.

The reason there's a try catach is to provide more context. Diagnostics are aribtrary functions. They also meant to be extended by users. This block tries to catch possible problems and point the user to the specific problem instead of having (just) a complex stacktrace. 

> Also, the block inside if config.parsed_args["warn_allocations_diagnostics"] doesn't really make sense, the contents will always allocate on the first call.

That's not a first call. The first call occurs in the try-catch.

Reply:
`diagnostics_iterations` won't exist, and you'll get an error when there's zero diagnostics.